data/migrations/20190423082317_create_users_table.js

```diff
@@ -1,6 +1,13 @@
 exports.up = function(knex, Promise) {
   return knex.schema.createTable('users', tbl => {
-    tbl.increments();
+    if (knex.client.config.client === 'pg') {
+      tbl
+        .uuid('id')
+        .primary()
+        .defaultTo(knex.raw('uuid_generate_v4()'));
```

Comment:
I was envisioning the UUID to be generated in the JS files, then inserted into the database. If I understand this correctly, knex can generate UUID's on its own?

Reply:
Postgres has a built-in UUID type, and you can use the uuid-ossp module to generate a uuid in this way.